internal/faustjs.org/docs/gutenberg/tutorial/create-a-custom-block.mdx

```diff
@@ -0,0 +1,314 @@
+
+## 2. Defining your custom block
+
+In this tutorial, we will assume that you do not have a preexisting block in mind to use with your Decoupled site. If you have a block you want to emulate from the WordPress core blocks library, head to our [tutorial](/docs/gutenberg/tutorial/create-a-block-from-wordpress-core.mdx) detailing how to do exactly that. Likewise, if you seek to recreate a block used by a third party plugin, we have a [tutorial](/docs/gutenberg/tutorial/create-a-block-from-third-party.mdx) tailored for that process, too.
+
+This tutorial assumes that you begin your block creation from nothing at all, which will therefore require a few extra steps to work. 
+
+:::tip 
+
+Check out all the components that are available to you through the [@wordpress/components](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-components/) library as you think about what you want your custom block to do and look like. 
+
+:::
+
+## 3. Creating your block using the WordPress block boilerplate code
+
+The [@wordpress/create-block](https://www.npmjs.com/package/@wordpress/create-block) package exists to create the necessary block scaffolding to get you started. See the WordPress basic block [tutorial](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/) documentation for a step-by-step guide.
+
+For this step, we will make a simple block named MyCoolBlock, which has a RichText attribute and a radial button attribute from the @wordpress/components library.
+
+## 4. Create the initial block
+
+Inside the codebase of your application, create the basic block structure that simply prints the block's attributes.
+The steps required to achieve that are the following:
+
+### 4.1 Save it inside the `wp-blocks` folder
+
+```jsx title="wp-blocks/DnDBlock.js"
+export default function MyCoolBlock(props) {
+  console.log(props.attributes);
+  return <div>MyCoolBlock</div>
+}
+```
+
+This component prints the block attributes via the props passed as a parameter.
+
+### 4.2 Export the block in the `index.js`
+
+```jsx title="wp-blocks/index.js"
+import MyCoolBlock from './MyCoolBlock';
+export default {
+  MyCoolBlock
+}
+```
+
+By exporting the block in the `index.js` we make it available in the `WordPressBlocksProvider`.
+
+### 4.3 Create the block fragment
+
+You want to create a fragment that describes the request block fields and attributes. Ideally you want to include all the relevant fields for the implementation part:
+
+```jsx title="wp-blocks/MyCoolBlock.js" {1}
+import {gql} from '@apollo/client';
+...
+MyCoolBlock.fragments = {
+  key: `MyCoolBlockBlockFragment`,
+  entry: gql`
+    fragment MyCoolBlockBlockFragment on MyCoolBlock {
+      attributes {
+        class
+        content
```

Comment:
Maybe you should use only the `content` field since the other fields are not really used.

Reply:
For my example block, I do have color and message added. I may fiddle with this a little later though and ask for one more round of feedback. Thank you!